### ISSUE TITLE
Docs need more info on `#:wrapper` functions

### ISSUE THREAD

**Issue by otherjoel** [Bogdanp/deta#12]
There isn’t much info on the docs on the precise behavior of `#:wrapper`-specified functions. The tutorial has one examples of it in use, but there is actually no description of its behavior or capabilities under the docs `define-schema` where one would expect to find it.

For example, at first I thought I could use it to automatically convert a value to the required type, such as for dates:

```racket
#lang racket/base

(require deta db gregor)

(define-schema article
  ([id id/f #:primary-key #:auto-increment]
   [date date/f #:wrapper iso8601->date]))

(define cache-conn (sqlite3-connect #:database "db.sqlite" #:mode 'create))

(create-table! cache-conn 'article)

(define a (make-article #:date "2020-01-04"))

(insert-one! cache-conn a)
```

But this results in an error:

```
make-article: contract violation
  expected: date-provider?
  given: "2020-01-04"
  in: an and/c case of
      the #:date argument of
      (->*
       (#:date (and/c date-provider? any/c))
       (#:id (and/c natural? any/c))
       article?)
  contract from: (function make-article)
  blaming: /Users/joel/Documents/code/sandbox/deta/deta-1.rkt
   (assuming the contract is correct)
  at: /Users/joel/Documents/code/sandbox/deta/deta-1.rkt
```

I’m not saying this should be allowed; rather, that it would be good for the docs for `define-schema` to explain exactly what `#:wrapper` functions can and can’t do. If you will accept a pull request for this I will be happy to work on it.

**Comment by Bogdanp** [comment]
> If you will accept a pull request for this I will be happy to work on it.

I'd be happy to!

The relevant code is [here](https://github.com/Bogdanp/deta/blob/56f0eb108a8f4c4f73ea021a9823557c5d4a0c87/deta-lib/schema.rkt#L40-L53) and the intended behavior is to normalize values, without changing their type.  A common use case for me is ensuring usernames are always lower-cased:

```racket
(define-schema user
  ([username string/f #:wrapper string-downcase]))
```

**Comment by Bogdanp** [comment]
Closed in #13. Thanks again!
